src/server/service/achievements.ts

```diff
@@ -1,46 +1,111 @@
+  description: "Convince 2 or more humans that you are a bot in a match",
+  calculate: ({ otherPlayers, player }) => {
+    let isVotedAgainst = 0;
+    otherPlayers.forEach((p) => {
+      if (p.votes?.includes(player.userId)) isVotedAgainst += 1;
+    });
+    return isVotedAgainst >= 2;
+  },
+};
+
+const busterStreakAchievement: Achievement = {
+  name: "Buster Streak",
+  description: "Bust three bots in a row",
+  calculate: ({ player, playerHistory, otherPlayers, botsBusted }) => {
+    if (!playerHistory || playerHistory.length < 2) return false;
+    if (alreadyReceivedAchievementToday(player.userId, playerHistory, "101"))
+      return false;
+
+    // Sort playerHistory in descending order by timestamp
+    const sortedPlayerHistory = playerHistory.sort(
+      (a, b) => b.createdAt - a.createdAt
+    );
+
+    // Get the last two matches
+    const lastTwoMatches = sortedPlayerHistory.slice(0, 2);
+
+    //Get the player data from the last two matches
+    const playerMatchData = lastTwoMatches.map((match) => {
+      return match.players.find((p) => p.userId === player.userId);
+    });
+
+    // Check if all bots were busted in the last two matches
+    const allBotsBustedLastTwoMatches = playerMatchData.every((p) => {
+      const perfectScore = p?.achievements.filter((achievement) => {
+        return achievement === "12";
+      });
+      return perfectScore && perfectScore.length > 0;
+    });
+
+    // Get the bots from the current match
+    const agents = otherPlayers.filter((p) => p.isBot);
+
+    const perfectCurrentGame = botsBusted === agents.length;
+
+    return allBotsBustedLastTwoMatches && perfectCurrentGame;
+  },
+};
+
+const firstTimerAchievement: Achievement = {
+  name: "First-Timer",
+  description: "Played your first match",
+  calculate: ({ playerHistory }) => {
+    if (!playerHistory || playerHistory.length === 0) return true;
+    return false;
+  },
+};
+
+const beginnersLuckAchievement: Achievement = {
+  name: "Beginners Luck",
+  description: "Player busts at least one bot in the first match",
+  calculate: ({ playerHistory, botsBusted }) => {
+    const isFirstMatch = !playerHistory || playerHistory.length === 0;
+
+    return isFirstMatch && botsBusted > 0;
   },
 };
 
-export const MATCH_ACHIEVEMENTS: Record<string, MatchAchievement> = {
-  // Match achievement - written last message
-  "11": lastMessageAchievement,
-  // Match achievement - perfect score (all votes correct)
-  // "12": perfectScoreAchievement,
-  // Match achievement - someone selected you as a bot
-  // "13": 10,
-  // Day achievement - say a specific word
-  // "101": 5,
-  // Day achievement - successfully bust all bots 3 consecutive games
-  // "102": 10,
-  // Day achievement - Daily streak plays bot busters X days in a row
-  // "103": 10,
-  // One time achievement - player wins his first game
-  // "201": 10,
+export const MATCH_ACHIEVEMENTS: Record<AchievementId, Achievement> = {
+  "11": lastOneAchievement,
```

Comment:
If all our IDs are going to be numbers, we may as well define the proper type as num

Reply:
Yeah I was having a hard time deciding, how I want to structure the ID and points of the achievements. I think we can have a discussion of how we want to approach this because I am also not a big fan of how I made it now....